### PR TITLE
Do not list concealed pages

### PR DESCRIPTION
**Pull request by jage**
* Page method `with_concealed_if(bool)` to list concealed pages
* Move Page.search to dataset extension
* Show a lock for concealed pages in lists

### CODE REVIEW

File: lib/controllers/page_controller.rb [hunk 1]
```diff
@@ -30,13 +30,16 @@ def restrict_concealed(page)
...
+    @page_groups = Page.order(:title_char, :title).
+      with_concealed_if(starkast?).
+      to_hash_groups(:title_char)
     haml :index
   end
 
   get '/latest' do
     @page_title = "Senast ändrad"
     @page_groups = Page.order(:updated_on).reverse.
+      with_concealed_if(starkast?).
```
Comment (jage) [inline lib/controllers/page_controller.rb:42]: How about we change to dot first style @dentarg ?

File: views/page/index.haml [hunk 2]
```diff
@@ -11,5 +11,7 @@
     - pages.each do |page|
       %li
         %a{ href: "#{page.slug}" }= page.title
+        - if page.concealed
+          🔐
```
Comment (jage) [inline views/page/index.haml:15]: Mostly to make it easier when playing around with this, WDYT, keep? @dentarg

**Review by jage — COMMENTED** [review]
Maybe we can do a refactor round to make it nicer, but think this works.

Tried to make concealed excluded by default, but tricky to remove the exclusion on demand.

File: lib/controllers/page_controller.rb (hunk 1 above)
Comment (dentarg) [reply lib/controllers/page_controller.rb:42]: Sure

File: views/page/index.haml (hunk 2 above)
Comment (dentarg) [reply views/page/index.haml:15]: Sure, some emoji is always nice

**Review by dentarg — APPROVED** [review]
LGTM

File: lib/controllers/page_controller.rb (hunk 1 above)
Comment (jage) [reply lib/controllers/page_controller.rb]: Fixed in da5a0b9d9d4244b9127ec2826b246bc96ffa7f2c

**Comment by jage** [comment]
Added some tests as well, merging now.

File: lib/controllers/page_controller.rb [hunk 3]
```diff
@@ -85,7 +88,7 @@ def restrict_concealed(page)
   end
 
   get '/:slug/edit' do
-    @page = Page.find(slug: slug)
+    @page = Page.with_concealed_if(starkast?).find(slug: slug)
```
Comment (dentarg) [inline lib/controllers/page_controller.rb:91]: I assume this was changed by mistake, had to change it back in https://github.com/Starkast/wikimum/commit/257af591af3951368457e7586b32c064db112d7a

Error was

```
2022-04-22 10:52:14 - NoMethodError - undefined method `title' for #<Enumerator:0x00007fab9072d830>:
/app/lib/controllers/page_controller.rb:92:in `block in <class:PageController>'
```

We now have tests for this route :-)

Comment (jage) [reply lib/controllers/page_controller.rb:91]: Correct.

> We now have tests for this route :-)

👍

**Pull request merged**